tests/Unit/DataRowTest.php

```diff
@@ -171,6 +175,38 @@ public function testValidationOfEmptyDataRow()
         $dataItemManipulator = function () {
         };
 
-        $dataRow->prepare($keyFields, $fieldMap, $dataItemManipulator);
+        $dataRow->prepare([], $keyFields, $fieldMap, $dataItemManipulator);
+    }
+
+    public function testValidationRulesFailure()
+    {
+        $this->expectException(ValidationException::class);
```

Comment:
I think it should have the `$this->expectExceptionMessage` to check whether the expected exception is same as actual.